### PR TITLE
[InProgress] Basic web app creation

### PR DESCRIPTION
**Pull request by jianghaolu**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@jianghaolu__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: azure-mgmt-website/src/main/java/com/microsoft/azure/management/website/implementation/SiteImpl.java [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        return this;
+    }
+
+    @Override
+    public Site.DefinitionStages.WithCreate withNewAppServicePlan() {
+        return null;
+    }
+
+    @Override
+    public Site.DefinitionStages.WithCreate withNewAppServicePlan(String name, AppServicePricingTier pricingTier) {
```
Comment (martinsawicki) [inline azure-mgmt-website/src/main/java/com/microsoft/azure/management/website/implementation/SiteImpl.java]: generally, whenever we can, return the impl itself, not the specific interface the impl is implementing, that way we separate the user-facinmg logic from the impl and enable ourselves to use the same impl'd method to implement multiple flulent interfaces (Definition & Update) in one go. So these withers should consistently return SiteImpl.

File: azure-mgmt-website/src/main/java/com/microsoft/azure/management/website/implementation/SiteImpl.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+        return null;
+    }
+
+    @Override
+    public Site.DefinitionStages.WithCreate withExistingAppServicePlan(String appServicePlanName) {
```
Comment (martinsawicki) [inline azure-mgmt-website/src/main/java/com/microsoft/azure/management/website/implementation/SiteImpl.java]: same as before

File: azure-mgmt-website/src/main/java/com/microsoft/azure/management/website/implementation/SitesImpl.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        return new SiteImpl(name, new SiteInner(), innerCollection, super.myManager);
+    }
+
+    @Override
+    protected SiteImpl wrapModel(SiteInner inner) {
+        return new SiteImpl(inner.name(), inner, innerCollection, super.myManager);
+    }
+
+    @Override
+    public Site.DefinitionStages.Blank define(String name) {
```
Comment (martinsawicki) [inline azure-mgmt-website/src/main/java/com/microsoft/azure/management/website/implementation/SitesImpl.java]: similar issue - it's the interface that determines the return value, so the impl can just return SiteImpl.

File: azure-mgmt-website/src/main/java/com/microsoft/azure/management/website/WebApp.java [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+            WithCreate withExistingAppServicePlan(String appServicePlanName);
+        }
+
+        interface WithHostNameSslStates {
+            WithCreate disableSsl(String hostName);
+            WithCreate enableSniSsl(String hostName, String thumbprint);
+            WithCreate enableIpBasedSsl(String hostName, String thumbprint, String virtualIp);
+        }
+
+        interface withSiteEnabled {
```
Comment (martinsawicki) [inline azure-mgmt-website/src/main/java/com/microsoft/azure/management/website/WebApp.java]: oops -- capitalization mistake - interface names should be cap'd.

File: azure-mgmt-website/src/main/java/com/microsoft/azure/management/website/WebApp.java [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+            WithCreate siteEnabled(boolean enabled);
+        }
+
+        interface withScmSiteAlsoStopped {
+            WithCreate alsoStopScmSiteWhenStopped(boolean scmSiteAlsoStopped);
+        }
+
+        interface withClientAffinityEnabled {
```
Comment (martinsawicki) [inline azure-mgmt-website/src/main/java/com/microsoft/azure/management/website/WebApp.java]: another one -- and a couple around it, so I won't comment on each.

**Comment by jianghaolu** [comment]
Blocked by Web app swagger rework. Will reopen later this month.

**Pull request closed without merging**